test/framework/rootpod_executor.go

```
@@ -71,15 +71,33 @@ func (e *rootPodExecutor) Execute(ctx context.Context, command ...string) ([]byt
 	stdout, stderr, err := e.client.PodExecutor().Execute(ctx, e.Pod.Namespace, e.Pod.Name, e.Pod.Spec.Containers[0].Name,
 		append([]string{"chroot", "/hostroot"}, command...)...,
 	)
-	if err != nil {
-		return nil, err
-	}
 
+	var stderrBytes, stdoutBytes []byte
 	if stderr != nil {
-		return io.ReadAll(stderr)
```

Why was this changed, the former version was concise and looked good?

---

In the new pod executor implementation the reader is never nil, but returns a zero byte array.

---

Hm, I don't understand. Shouldn't it be fine to return an empty byte array as long as the error is returned and handled by the caller?

---

If `stderr` is empty, it should return `stdout`. If `stdout` is empty too, it should return an error.

Otherwise, the check cannot validate the output of the command which was executed.